### PR TITLE
Misc updates

### LINKED ISSUE #1: Option for runner to register with a specific job tag
**Is your feature request related to a problem? Please describe.**

It would be great if you could optionally pass in the tag(s) the runner to register with in the stack configuration.

Detail:

In GitLab CI you can configure runners (or coordinators in this case) with a tag(s) so that they only pick-up CI jobs with that tag(s), see 
https://docs.gitlab.com/ee/ci/runners/#using-tags and https://docs.gitlab.com/ee/ci/yaml/README.html#tags

This is extremely useful for separating stages that may need to run in or deploy to different environments.

For example you might have separate runners deployed in your pre-prod and prod environments, production deployments might be run by a stage with a tag that only production agents pick-up.

When registering a runner manually you specify the specific tags during `gitlab-runner register` process:

```
# Please enter the gitlab-ci tags for this runner (comma separated):
my-tag,another-tag
```

**Describe the solution you'd like**

I started to look into this and I can see it might be a bit of a pain as the register command that would need the tag is baked into the container image, maybe it could be read from an environment variable?

--

_also thank you for taking the time to share your pattern for this!_

### PR DESCRIPTION
**Pull request by GMartinez-Sisti**
Hi!

This repo is really useful and helped me a lot.

Here are some suggestions that hopefully will be useful :)

- Secure GitlabRegistrationToken with AWS Secrets Manager
     - Remove token from Cloudformation templates and ECS Task config
- Remove trailing whitespace
    - This was made by my VSCode I think it's worth keeping it :)
- Add runner tag list option
    - Add option to specify runner tags from the Cloudformation template (closes https://github.com/woodjme/autoscaling-ec2-gitlab-runners-fargate/issues/1)
- Update additional parameters on entrypoint
    - Ensure bash best practices
- Unregister gitlab-runner on exit
    - This will make the runner unregister itself when it ends/crashes/recycles
- Set error control in entrypoint
    - Based on bash best practices also
- Use docker machine version recommended by gitlab
    - Docker recommends its own version of docker machines ([here](https://docs.gitlab.com/runner/executors/docker_machine.html#forked-version-of-docker-machine))
- Add dumb-init to docker image
    - Ensure handling of processes and signals ([Dumb-init repository](https://github.com/Yelp/dumb-init)) 
- Decrease docker image size
    - Less layers by joining commands
- Allow specifying the number of concurrent VMs to spawn
    - When deploying the stack we can now customize this value

I am able to start/register/unregister the container locally. Will also try with an upload to ECR and start it in ECS to test it fully in AWS.

Regards

### CODE REVIEW

**Comment by GMartinez-Sisti** [comment]
Just added `Allow specifying the number of concurrent VMs to spawn`. Won't change anything else!

I'm now testing the stack with a custom image that includes these changes.

**Comment by woodjme** [comment]
Created a `dev` branch with corresponding `dev` tagged docker image.

[Dev CF Template](https://autoscaling-ec2-gitlab-runners-fargate.s3-eu-west-1.amazonaws.com/dev/gitlab-runner-template.yml)

File: Dockerfile [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-chmod +x /tmp/docker-machine && \
-cp /tmp/docker-machine /usr/local/bin/docker-machine
+	dumb-init && \
+	# Decrease docker image size
+	rm -rf /var/lib/apt/lists/* && \
+	# Install Gitlab Runner
+	curl -LJO https://gitlab-runner-downloads.s3.amazonaws.com/v${GITLAB_RUNNER_VERSION}/deb/gitlab-runner_amd64.deb && \
+	dpkg -i gitlab-runner_amd64.deb && \
+	# Install Docker Machine
+	curl -L https://gitlab-docker-machine-downloads.s3.amazonaws.com/v{DOCKER_MACHINE_VERSION}/docker-machine > /usr/local/bin/docker-machine && \
```
Comment (GMartinez-Sisti) [inline Dockerfile]: I forgot the `$` in the variable here :( I'll send a fix for it.

**Comment by GMartinez-Sisti** [comment]
@woodjme Deploying the stack with a custom image worked perfectly.

This time for real I won't change anything else :)

Thanks

**Pull request merged**